lib/gssapi/mech/gss_acquire_cred_from.c

```diff
@@ -166,17 +166,29 @@ gss_acquire_cred_from(OM_uint32 *minor_status,
 	int match = 0;
 
 	for (i = 0; i < desired_mechs->count; i++) {
-	    gss_test_oid_set_member(minor_status, &desired_mechs->elements[i],
-				    _gss_mech_oids, &match);
-	    if (match)
+	    m = __gss_get_mechanism(&desired_mechs->elements[i]);
+	    if (m != NULL) {
+		/*
+		 * Now SPNEGO supports GM_USE_MG_CRED it's no longer necessary
+		 * to specifically acquire SPNEGO credentials. However a legacy
+		 * application will do this with the intention of negotiating
+		 * all available mechanisms (subject to refinement by calling
+		 * GSS_Set_neg_mechs()). Interpret GM_USE_MG_CRED to mean all
+		 * mechanisms.
+		 */
+		if (m->gm_flags & GM_USE_MG_CRED)
+		    mechs = _gss_mech_oids;
```

Comment:
I think we should do this IFF there's only `GM_USE_MG_CRED` mechs in `desired_mechs`.  That's why my version of this has an `only_neg_mechs` variable.

Reply:
Will fix.

Reply:
Fixed in current version.

Reply:
This we resolved on IM last night with proposed new code from @lhoward.